### PR TITLE
Support extra ops modes for LLM Models

### PR DESCRIPTION
**Pull request by Jiseong-oh**
### Summary
Add Extra Ops for Exynos Backend
- Cos, GroupNorm, Index, Log, Pow, Rms_Norm, Sigmoid, sing, Splite_with_sizes_copy, sum_int_list, tanh and Topk
- All ops are verfied
- These ops will be used for supporting LLM models

Support Perf mode
- It can be supported Perf mode with experimental
- This mode MUST BE used to verify model on exynos device farm fristly before testing on the phone

cc @SS-JIA @digantdesai @kimishpatel

### CODE REVIEW

**Comment by SS-JIA** [comment]
@claude help review this PR

**Review by SS-JIA — APPROVED** [review]
Stamp to unblock -- but the agent review comments are valid. Would highly recommend fixing them before landing!

File: backends/samsung/builders/op_topk.py [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+        all_output_tensors = []
+        users = list(node.users.keys())
+        output_val_idx = 0
+        output_val_id = self.define_tensor(
+            node,
+            enn_graph,
+            vals_to_ids,
+            output_idx=output_val_idx,
+        )
+        if len(users) > 0 and users[0].target.__name__ == "getitem":
```
Comment (SS-JIA) [inline backends/samsung/builders/op_topk.py:49]: **High — TopK output mapping uses user order rather than tuple index.**

This maps `users[0]` to values and `users[1]` to indices without checking each `getitem` index. For `torch.topk(x, k)[1]`, the only user is mapped to output 0, so callers can receive values where indices were requested. Reversed and partial tuple use have the same problem; output-boundary classification can also disagree with the compiled ENN output count.

Please map each `getitem` through `user.args[1]`, preserve output-specific boundary status, and cover values-only, indices-only, reversed-use, and downstream-consumer cases.

_AI reviewer disclosure: This comment was generated by an AI reviewer._

File: backends/samsung/builders/op_split_with_sizes_copy.py [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+        enn_graph: EnnGraph,
+        vals_to_ids: Dict[torch.Tensor, int],
+    ) -> None:
+        input = node.args[0]
+        input_id = self.define_tensor(input, enn_graph, vals_to_ids)
+
+        # output
+        all_output_tensors = []
+
+        for output_idx in range(len(node.args[1])):
```
Comment (SS-JIA) [inline backends/samsung/builders/op_split_with_sizes_copy.py:33]: **High — Split outputs are not associated with their `getitem` consumers.**

Each output is created by repeatedly associating an ID with the tuple-producing node, while the registered `getitem` visitor is a no-op. In a graph such as `a, b = torch.split(...); return a + b`, the add visitor can create fresh, producerless tensors for `a` and `b`, disconnecting them from `SPLIT`. Returning only a subset can also make delegate and compiled output counts disagree.

Please map each result ID to the corresponding indexed `getitem` user and add downstream-consumer and subset-output tests.

_AI reviewer disclosure: This comment was generated by an AI reviewer._

File: backends/samsung/builders/op_group_norm.py [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+        bias_node = node.args[2]
+        bias_id = self.define_tensor(bias_node, enn_graph, vals_to_ids)
+        all_input_tensors.append(bias_id)
+
+        num_groups = cast(int, node.args[6])
+        epsilon = node.args[7]
+
+        params = {"num_groups": num_groups, "epsilon": epsilon}
+
+        output_id = self.define_tensor(node, enn_graph, vals_to_ids, output_idx=0)
```
Comment (SS-JIA) [inline backends/samsung/builders/op_group_norm.py:43]: **High — GroupNorm output is not connected to a consuming op.**

`aten.native_group_norm` returns a tuple, but this serializes output 0 without mapping its `getitem(0)` user. A graph such as `relu(group_norm(x))` therefore lets the ReLU visitor create a new orphan tensor instead of consuming the `GROUPNORM` output. The isolated test misses this because GroupNorm terminates at model output.

Please map `getitem(0)` to `output_id`, explicitly reject unsupported mean/rstd uses, and add a GroupNorm-followed-by-op test.

_AI reviewer disclosure: This comment was generated by an AI reviewer._

File: backends/samsung/_passes/compose_rms_norm.py [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+        partitions = get_source_partitions(graph, [torch.nn.RMSNorm])
+
+        for _, src_partitions in partitions.items():
+            for src_partition in src_partitions:
+                input_len = len(src_partition.input_nodes)
+                if input_len == 1:
+                    input_node = src_partition.input_nodes[0]
+                elif input_len == 2:
+                    inp_0, inp_1 = src_partition.input_nodes
+                    input_node = inp_0 if len(inp_0.users) == 2 else inp_1
```
Comment (SS-JIA) [inline backends/samsung/_passes/compose_rms_norm.py:66]: **High — RMSNorm input selection relies on unstable ordering and global user counts.**

`get_source_partitions()` does not provide semantic ordering for `input_nodes`, and `users` includes uses outside the matched partition. Common pre-norm residual graphs add an external use of the activation; shared gamma parameters can similarly change counts. This heuristic can therefore choose gamma as the RMSNorm input and produce invalid shapes or wrong results.

Please identify activation and gamma structurally from the matched RMSNorm expression, with residual-input and shared-parameter tests.

_AI reviewer disclosure: This comment was generated by an AI reviewer._

File: backends/samsung/builders/op_group_norm.py [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+        self,
+        node: torch.fx.Node,
+        enn_graph: EnnGraph,
+        vals_to_ids: Dict[torch.Tensor, int],
+    ) -> None:
+        all_input_tensors = []
+        input_id = self.define_tensor(node.args[0], enn_graph, vals_to_ids)
+        all_input_tensors.append(input_id)
+
+        weight_node = node.args[1]
```
Comment (SS-JIA) [inline backends/samsung/builders/op_group_norm.py:31]: **Medium — Valid non-affine GroupNorm is claimed as supported but fails lowering.**

For `torch.nn.GroupNorm(..., affine=False)`, weight and bias are both `None`. Passing them unconditionally to `define_tensor` reaches tensor metadata access and fails preprocessing. Since registration accepts every `native_group_norm.default`, this errors instead of falling back.

Please support absent affine inputs or reject this configuration during partitioning, and add an `affine=False` test.

_AI reviewer disclosure: This comment was generated by an AI reviewer._

File: examples/samsung/scripts/resnet18.py [hunk 6]
```diff
@@ -144,7 +145,9 @@ def get_data_loader():
     test_in = inputs[0]
     float_out = model(*test_in)
 
-    compile_specs = [gen_samsung_backend_compile_spec(args.chipset)]
+    compile_specs = [
+        gen_samsung_backend_compile_spec(args.chipset, PerformanceMode.HIGH_PERFORMANCE)
```
Comment (SS-JIA) [inline examples/samsung/scripts/resnet18.py:149]: **Low — Standalone examples force experimental high-performance mode.**

This script and ten sibling model scripts unconditionally choose `HIGH_PERFORMANCE`, despite the PR's device-farm validation requirement. The generic compiler now safely defaults to `DEFAULT` and exposes `--perf_mode`; standalone scripts should expose the same choice or retain `DEFAULT`. The README wording should also match actual defaults.

_AI reviewer disclosure: This comment was generated by an AI reviewer._
